src/commands/CmdImport.cpp

```diff
@@ -142,19 +142,31 @@ int CmdImport::import (const std::string& input)
   //   { ... }
   catch (std::string& e)
   {
+    // Make a very cursory check for the old-style format.
+    if (input[0] != '{') {
+      throw e;
+    }
+
+    // Read the entire file, so that errors do not result in a partial import.
+    std::vector<std::unique_ptr<json::object>> objects;
     for (auto& line : split (input, '\n'))
     {
       if (line.length ())
       {
         json::value* root = json::parse (line);
-        if (root)
-        {
-          importSingleTask ((json::object*) root);
-          ++count;
-          delete root;
-        }
+        if (root && root->type () == json::j_object)
+          objects.push_back (std::unique_ptr<json::object> ((json::object *)root));
+        else
+          // Throw the existing error, as it's likely more accurate.
+          throw e;
```

Comment:
My thought was that it would no longer be necessary to throw an error from the new-style parsing in the middle of old-style parsing. Seems like we should commit to a given format and act accordingly. Maybe just `throw format("Invalid Json: {1}", line)`?